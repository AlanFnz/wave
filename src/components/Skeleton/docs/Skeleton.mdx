---
name: Skeleton
menu: Components
route: /components/skeleton
---

import { Playground } from 'docz';
import { Skeleton } from '../Skeleton';
import { SkeletonPropsTable } from './SkeletonPropsTable';
import { StyledSystemLinks } from '../../../docs/StyledSystemLinks';
import { Combination } from '../../../docs/Combination';

# Skeleton

Provide a placeholder while you wait for content to load, or to visualise content that doesn't exist yet.

## Properties

<SkeletonPropsTable /><br/>

<StyledSystemLinks component="Skeleton" supportedProps={['layout', 'margin', 'borderRadius']} />

## Examples

<Playground>
    <Skeleton width="200px" animated />
</Playground>

## Combinations

<Combination animated={[false, true]}>
    {(props, i) => <Skeleton key={i} {...props} />}
</Combination>
