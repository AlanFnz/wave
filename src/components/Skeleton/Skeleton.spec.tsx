import { render } from '@testing-library/react';
import React from 'react';
import { Skeleton } from './Skeleton';

describe('Skeleton', () => {
    it('should render default', () => {
        const { container } = render(<Skeleton />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render animated version', () => {
        const { container } = render(<Skeleton animated />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
