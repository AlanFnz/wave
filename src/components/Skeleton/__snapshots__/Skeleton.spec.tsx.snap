// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render animated version 1`] = `
.c0 {
  display: inline-block;
  height: 1rem;
  width: 100%;
  position: relative;
  overflow: hidden;
  border-radius: 0.25rem;
  background-color: var(--wave-s-color-background-surface-neutral-faded);
}

.c0::after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-transform: translateX(-75%);
  -ms-transform: translateX(-75%);
  transform: translateX(-75%);
  background-image: linear-gradient( 90deg, rgba(255,255,255,0) 25%, rgba(255,255,255,0.5) 37%, rgba(255,255,255,0) 63% );
  background-repeat: no-repeat;
  content: '';
  -webkit-animation: ctGIgz 1600ms ease infinite;
  animation: ctGIgz 1600ms ease infinite;
  -webkit-animation-delay: 200ms;
  animation-delay: 200ms;
}

<span
  class="c0"
/>
`;

exports[`Skeleton should render default 1`] = `
.c0 {
  display: inline-block;
  height: 1rem;
  width: 100%;
  position: relative;
  overflow: hidden;
  border-radius: 0.25rem;
  background-color: var(--wave-s-color-background-surface-neutral-faded);
}

<span
  class="c0"
/>
`;
